src/TLuaInterpreter.cpp

```diff
@@ -10320,16 +9893,15 @@ int TLuaInterpreter::setDoor(lua_State* L)
                         luaL_typename(L, 3));
         lua_error(L);
         return 1;
-    } else {
-        doorStatus = lua_tointeger(L, 3);
-        if (doorStatus < 0 || doorStatus > 3) {
-            lua_pushnil(L);
-            lua_pushfstring(L,
-                            "setDoor: bad argument #3 value (door type %d is not one of 0=\"none\", 1=\"open\",\n"
-                            "2=\"closed\" or 3=\"locked\".)",
-                            doorStatus);
-            return 2;
-        }
+    }
+    doorStatus = lua_tointeger(L, 3);
```

Comment:
drop the declaration above and
```suggestion
    int doorStatus = lua_tointeger(L, 3);
```